### PR TITLE
Countdown improvement

### PR DESCRIPTION
**Pull request by kevincaradant**

### CODE REVIEW

**Comment by maxime1992** [comment]
It seems that the indentation of `backend/index.js` has changed.

Can you please re-indent as it was before to have a cleaner diff ?

I don't have any linter setup on the backend ATM and if you tried to run `ng lint` it was probably OK but it's only linting the frontend.

Thanks :)

**Comment by maxime1992** [comment]
Also, I've made few updates in the morning, could you rebase ?

File: frontend/src/app/shared/services/countdown.service.ts [hunk 1]
```diff
@@ -18,15 +18,18 @@ export class CountdownService {
           new Date(now.getFullYear(), now.getMonth(), now.getDate(), hour, minute, 0),
           ts => {
             // if minutes or seconds are < 10 add a 0 before
```
Comment (maxime1992) [inline frontend/src/app/shared/services/countdown.service.ts]: This comment needs an update as we'll have the hours too

`// if minutes, seconds or hours are < 10 add a 0 before`

Comment (kevincaradant) [inline frontend/src/app/shared/services/countdown.service.ts]: OK I'll change that ASAP ;)

**Comment by kevincaradant** [comment]
Ok, I'll fix the indentation of the backend file and rebase with your latest commits :)

File: frontend/src/app/shared/services/countdown.service.ts [hunk 2]
```diff
@@ -18,15 +18,18 @@ export class CountdownService {
...
-            const time = `${('0' + ts.minutes).slice(-2)}:${('0' + ts.seconds).slice(-2)}`;
+            const hours = ('0' + Math.floor(ts.minutes / 60)).toString().slice(-2);
+            const minutes = ('0' + ts.minutes % 60).toString().slice(-2);
+            const seconds = ('0' + ts.seconds).toString().slice(-2);
```
Comment (maxime1992) [inline frontend/src/app/shared/services/countdown.service.ts]: As we're doing 3 times the same thing here, can you create a function into `frontend/src/app/shared/helpers/time.helper.ts` (create the file)
`function getFormatedTime(): { hours: string, minutes: string, seconds: string };`

and then use it like that : 

`const { hours, minutes, seconds } = getFormatedTime();`

File: backend/index.js [hunk 3]
```diff
@@ -12,123 +12,127 @@ const { PizzaDeLOrmeau } = require('./pizzas-providers/parse-pizza-de-l-ormeau')
...
+    minute: appCtrl.getOrdersModel().getMinuteEnd()
   })
 
+  callback()
+})
+
+vorpal
+.delimiter('pizza-sync$')
+.show();
+})
```
Comment (maxime1992) [inline backend/index.js]: New line needed

File: backend/index.js [hunk 4]
```diff
@@ -12,123 +12,127 @@ const { PizzaDeLOrmeau } = require('./pizzas-providers/parse-pizza-de-l-ormeau')
 const pizzaProvider = new PizzaDeLOrmeau()
 const appCtrl = new AppController(pizzaProvider)
 
+const dataNow = new Date();
```
Comment (maxime1992) [inline backend/index.js]: `currentDate` might be more relevant ?

File: frontend/src/app/shared/helpers/time.helper.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export function getFormatedTime(ts: { hours: number, minutes: number, seconds: number }): string {
+  // if minutes or seconds are < 10 add a 0 before
```
Comment (maxime1992) [inline frontend/src/app/shared/helpers/time.helper.ts]: hours too

File: frontend/src/app/shared/helpers/time.helper.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+  const hours = ('0' + ts.hours).toString().slice(-2);
+  const minutes = ('0' + ts.minutes).toString().slice(-2);
+  const seconds = ('0' + ts.seconds).toString().slice(-2);
+  return `${hours}:${minutes}:${seconds}`;
```
Comment (maxime1992) [inline frontend/src/app/shared/helpers/time.helper.ts]: Jump line

File: frontend/src/app/shared/services/countdown.service.ts [hunk 7]
```diff
@@ -16,25 +17,24 @@ export class CountdownService {
...
+            const time = getFormatedTime(ts);
 
             // tell listeners that a new value is available
             if (previousSecond && previousSecond > ts.seconds) {
               observer.next(time);
             }
 
             // stop the countdown if 0:0 or if already finished
-            if ((ts.minutes === 0 && ts.seconds === 0) ||  previousSecond && previousSecond < ts.seconds) {
+            if ((ts.minutes === 0 && ts.seconds === 0) || previousSecond && previousSecond < ts.seconds) {
```
Comment (maxime1992) [inline frontend/src/app/shared/services/countdown.service.ts]: I think that if you have for example `02:01:00` left, at `02:00:00` the countdown will stop

**Comment by maxime1992** [comment]
Seems nearly perfect to me :D !

Can you add just a line in the readme to say that by default the server starts with a countdown left to 1 hour but still can be changed on the fly with command line ?

**Pull request merged**